tests/unit/systems/docker/testing_images_test.py

```diff
@@ -0,0 +1,123 @@
+
+    @patch('dusty.systems.docker.testing_image.get_docker_client')
+    @patch('dusty.systems.docker.testing_image._make_installed_testing_image')
+    def test_ensure_image_exists_no_force_1(self, fake_make_installed_image, fake_get_docker_client):
+        fake_docker_client = Mock()
+        fake_get_docker_client.return_value = fake_docker_client
+        fake_docker_client.images.return_value = []
+        testing_spec = {'once': 'npm install'}
+        new_image_name = 'dusty/image'
+        ensure_image_exists(testing_spec, new_image_name)
+        fake_make_installed_image.assert_has_calls([call(testing_spec, new_image_name, volumes=[])])
+
+    @patch('dusty.systems.docker.testing_image.get_docker_client')
+    @patch('dusty.systems.docker.testing_image._make_installed_testing_image')
+    def test_ensure_image_exists_no_force_2(self, fake_make_installed_image, fake_get_docker_client):
+        fake_docker_client = Mock()
+        fake_get_docker_client.return_value = fake_docker_client
+        fake_docker_client.images.return_value = []
+        testing_spec = {'once': 'npm install'}
+        new_image_name = 'dusty/image'
+        ensure_image_exists(testing_spec, new_image_name, volumes=['os/path:contianer:path'])
+        fake_make_installed_image.assert_has_calls([call(testing_spec, new_image_name, volumes=['os/path:contianer:path'])])
+
+    @patch('dusty.systems.docker.testing_image.get_docker_client')
+    @patch('dusty.systems.docker.testing_image._make_installed_testing_image')
+    def test_ensure_image_exists_no_force_3(self, fake_make_installed_image, fake_get_docker_client):
+        fake_docker_client = Mock()
+        fake_get_docker_client.return_value = fake_docker_client
+        fake_docker_client.images.return_value = [{'RepoTags': ['dusty', 'dusty/dog']},
+                                                  {'RepoTags': ['dusty/images', 'dusty/image']}]
+        testing_spec = {'once': 'npm install'}
+        new_image_name = 'dusty/image'
+        ensure_image_exists(testing_spec, new_image_name)
+        fake_make_installed_image.assert_has_calls([])
+
+    @patch('dusty.systems.docker.testing_image.get_docker_client')
+    @patch('dusty.systems.docker.testing_image._make_installed_testing_image')
+    def test_ensure_image_exists_force(self, fake_make_installed_image, fake_get_docker_client):
+        fake_docker_client = Mock()
+        fake_get_docker_client.return_value = fake_docker_client
+        fake_docker_client.images.return_value = [{'RepoTags': ['dusty', 'dusty/dog']},
+                                                  {'RepoTags': ['dusty/images', 'dusty/image']}]
+        testing_spec = {'once': 'npm install'}
+        new_image_name = 'dusty/image'
+        ensure_image_exists(testing_spec, new_image_name, volumes=['os/path:contianer:path'],force_recreate=True)
+        fake_make_installed_image.assert_has_calls([call(testing_spec, new_image_name, volumes=['os/path:contianer:path'])])
+
+
+
+
+
```

Comment:
Could you please set up your editor to enforce having exactly one newline at the ends of files? Cuts down on diff noise.